lib/code_corps/policy/policy.ex

```diff
@@ -8,6 +8,25 @@ defmodule CodeCorps.Policy do
   alias CodeCorps.Policy
   alias Ecto.Changeset
 
+  @doc ~S"""
+  Determines if the specified user can perform the specified action on the
+  specified resource.
+
+  The resource can be a record, when performing an action on an existing record,
+  or it can be a map of parameters, when creating a new record.
+  """
+  @spec authorize(User.t, atom, map | struct) :: {:ok, :authorized} | {:error, :not_authorized}
+  def authorize(%User{} = user, action, resource) do
+    case user |> can?(action, resource) do
+      true -> {:ok, :authorized}
+      false -> {:error, :not_authorized}
+    end
+  end
+
+  @spec can?(User.t, atom, map | struct) :: boolean
+  defp can?(%User{} = user, :update, %Comment{} = comment), do: Policy.Comment.update?(user, comment)
+  defp can?(%User{} = user, :create, %{} = params), do: Policy.Comment.create?(user, params)
```

Comment:
As we switch controllers away from ja_resource, we also switch policies away from `Canary`. They also should no longer accept changesets and instead simply accept params maps.